src/Plugins/Reminder.php

```diff
@@ -0,0 +1,357 @@
+<?php declare(strict_types=1);
+
+namespace Room11\Jeeves\Plugins;
+
+use Amp\Promise;
+use Amp\Success;
+use Room11\Jeeves\Chat\Client\Chars;
+use Room11\Jeeves\Chat\Client\ChatClient;
+use Room11\Jeeves\Chat\Client\PostFlags;
+use Room11\Jeeves\Chat\Message\Command;
+use Room11\Jeeves\Chat\Room\Room as ChatRoom;
+use Room11\Jeeves\Storage\KeyValue as KeyValueStore;
+use Room11\Jeeves\Storage\Admin as AdminStore;
+use Room11\Jeeves\System\PluginCommandEndpoint;
+use function Amp\resolve;
+use function Amp\once;
+use function Amp\all;
+
+class Reminder extends BasePlugin
+{
+    private $chatClient;
+    private $storage;
+    private $watchers;
+    private $admin;
+
+    const USAGE = "Usage: `!!reminder [ examples | list | <text> [ at <time> | in <delay> ] | unset <id> ]` Try `!!reminder examples`" ;
+    const REMINDER_REGEX = "/(.*)\s+(?:in|at)\s+(.*)/ui";
+    const TIME_FORMAT_REGEX = "/(?<time>(?:\d|[01]\d|2[0-3]):[0-5]\d)[+-]?(?&time)?/ui";
+    const TIME_STRING_REGEX = <<<'REGEX'
+    /(?(DEFINE)
+      (?<int>
+        (\s*\b)?
+        (\d{1,5})?
+        (\s*)?
+      )
+      (?<timepart>
+        (?&int)
+          ( s(ec(ond)?s?)?
+          | m(in(ute)?s?)?
+          | h(rs?|ours?)?
+          | d(ays?)?
+          )
+          \b
+      )
+    )
+    
+    ^(?<time>(?:(?&timepart)(*SKIP).)+)(?<string>.+)$
+    /uix
+REGEX;
+
+
+    public function __construct(ChatClient $chatClient, KeyValueStore $storage, AdminStore $admin, array $watchers = []) {
+        $this->chatClient = $chatClient;
+        $this->watchers = $watchers;
+        $this->storage = $storage;
+        $this->admin = $admin;
+    }
+
+    private function flush(Command $command) {
+        return resolve(function() use($command) {
+            if (!yield $this->admin->isAdmin($command->getRoom(), $command->getUserId())) {
+                return $this->chatClient->postReply($command, "I'm sorry Dave, I'm afraid I can't do that");
+            }
+
+            $reminders = yield $this->storage->getKeys($command->getRoom());
+            if ($reminders !== []) {
+                foreach ($reminders as $key){
+                    yield $this->storage->unset($key, $command->getRoom());
+                }
+            }
+            return $this->chatClient->postMessage($command->getRoom(), "Reminders are gone.");
+        });
+    }
+
+    private function unset(Command $command) {
+        $messageId = $command->getParameter(1) ?? false;
```

Comment:
We can, in that case we would return if the `$messageId` is null (since `$command->getParameter()` would return null) instead of a false check.
Like:
```$messageId = $command->getParameter(1);
  return resolve(function() use($command, $messageId) {
      if($messageId === null){
          return $this->chatClient->postMessage($command->getRoom(), self::USAGE); 
      }
      // ...
});```